File: Bit Magic[13-May-2023]/Solution.java
//   >> JAVA CODE <<


public static int bitMagic(int n, int[] arr) {
        int low=0,high=arr.length-1;
        int cnt=0;
        while(low<high)
        {
            if(arr[low]!=arr[high])
                cnt++;
            
            low++; high--;
        }
        
        return cnt%2==0?cnt/2:cnt/2+1;
    }

File: Bit Magic[13-May-2023]/Solution.py
#  >> PYTHON CODE <<


def bitMagic(self, n : int, arr : List[int]) -> int:
    i,j=0,len(arr)-1
    
    cnt=0
    
    while(i<j):
        if(arr[i]!=arr[j]): cnt+=1
        i+=1
        j-=1
    
    if(cnt%2==0): return cnt//2 
    
    return cnt//2+1